FeedStoreChallenge/RealmFeedStore/RealmFeedStore.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import RealmSwift
+
+public final class RealmFeedStore: FeedStore {
+
+	public typealias Configuration = Realm.Configuration
+
+	private let realm: Realm
+
+	public init(configuration: Configuration = .defaultConfiguration) throws {
+		self.realm = try Realm(configuration: configuration)
+	}
+
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		do {
+			try realm.write {
+				realm.deleteAll()
+			}
+			completion(nil)
+		} catch {
+			completion(error)
+		}
+	}
+
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		let realmFeed = RealmFeed(
+			timestamp: timestamp,
+			realmFeedImages: feed.map(realmFeedImage)
+		)
+
+		do {
+			try realm.write {
+				realm.deleteAll()
+				realm.add(realmFeed)
+			}
+			completion(nil)
+		} catch {
+			completion(error)
+		}
+	}
+
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		guard let realmFeed = realm.objects(RealmFeed.self).first, !realmFeed.realmFeedImages.isEmpty else {
```

Comment:
An empty array of `realmFeedImages` is a valid cache - so the `isEmpty` logic shouldn't exist here.

```suggestion
		guard let realmFeed = realm.objects(RealmFeed.self).first else {
```